### PR TITLE
add neptune.ai logger hook

### PR DESCRIPTION
**Pull request by fcakyon**
## Motivation
We want to use mmdetection with neptune.ai. This PR adds support to neptune.ai logging.

## Modification
Added NeptuneLoggerHook

## BC-breaking (Optional)
No breaking changes

### CODE REVIEW

**Comment by fcakyon** [comment]
@ZwwWayne @xvjiarui i have created a new clean pr

File: mmcv/runner/hooks/logger/neptune.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Open-MMLab. All rights reserved.
+from ...dist_utils import master_only
+from ..hook import HOOKS
+from .base import LoggerHook
+
+
+@HOOKS.register_module()
+class NeptuneLoggerHook(LoggerHook):
+
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/neptune.py:9]: please add docstring

Comment (fcakyon) [reply mmcv/runner/hooks/logger/neptune.py:9]: @zhouzaida done.

File: mmcv/runner/hooks/logger/neptune.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            import neptune.new as neptune
+        except ImportError:
+            raise ImportError(
+                'Please run "pip install neptune-client" to install neptune')
+        self.neptune = neptune
+        self.run = None
+
+    @master_only
+    def before_run(self, runner):
+        if self.neptune is None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/neptune.py:62]: why we check the value of `self.neptune` again?

File: mmcv/runner/hooks/logger/neptune.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+class NeptuneLoggerHook(LoggerHook):
+
+    def __init__(self,
+                 init_kwargs=None,
+                 interval=10,
+                 ignore_last=True,
+                 reset_flag=True,
+                 with_step=True,
+                 by_epoch=True):
+        """Class to log metrics to NeptuneAI.
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/neptune.py:17]: Those docstring should be moved to line 9.

File: mmcv/runner/hooks/logger/neptune.py (hunk 2 above)
Comment (fcakyon) [reply mmcv/runner/hooks/logger/neptune.py:62]: i took wandb hooker as reference:
https://github.com/open-mmlab/mmcv/blob/b8c09f3b8449f2771e92a6a671027359d2c2707e/mmcv/runner/hooks/logger/wandb.py#L35

File: mmcv/runner/hooks/logger/neptune.py (hunk 3 above)
Comment (fcakyon) [reply mmcv/runner/hooks/logger/neptune.py]: done

File: mmcv/runner/hooks/logger/neptune.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+
+@HOOKS.register_module()
+class NeptuneLoggerHook(LoggerHook):
+    """Class to log metrics to NeptuneAI.
+
+    It requires `neptune-client` to be installed.
+
+    Args:
+        init_kwargs:
```
Comment (ZwwWayne) [inline mmcv/runner/hooks/logger/neptune.py:14]: The docstring should looks like this
```python
"""
init_kwargs (dict): a dict contains the initialization keys as below:

    - project (str): xxx
    - api_token (str): xxx
"""
```

Comment (fcakyon) [reply mmcv/runner/hooks/logger/neptune.py]: done

File: mmcv/runner/hooks/logger/neptune.py (hunk 2 above)
Comment (zhouzaida) [reply mmcv/runner/hooks/logger/neptune.py:63]: This line may be redundant. You could try again after removing the line.

Comment (fcakyon) [reply mmcv/runner/hooks/logger/neptune.py]: @zhouzaida done.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**